src/core/broker_config.h

```diff
@@ -43,18 +44,19 @@ class BrokerConfig
     kOk
   };
 
-  [[nodiscard]] ParseResult Read(const std::string& file_name);
-  [[nodiscard]] ParseResult Read(std::istream& stream);
+  [[nodiscard]] ParseResult Read(std::istream& stream, rdmnet::BrokerLog* log = nullptr);
+  void SetDefaults();
 
   [[nodiscard]] const etcpal::Uuid& default_cid() const { return default_cid_; }
 
   rdmnet::BrokerSettings settings;
+  int log_mask{ETCPAL_LOG_UPTO(ETCPAL_LOG_INFO)};
 
 private:
-  ParseResult ValidateCurrent();
+  ParseResult ValidateCurrent(rdmnet::BrokerLog* log);
```

Comment:
Any reason to have nodiscard on the other fn returning ParseResult and not on this one?